domain_access/domain_access.module

```diff
@@ -25,22 +26,27 @@ define('DOMAIN_ACCESS_USER_FIELD', 'field_domain_user');
 /**
  * Implements hook_node_grants().
  */
-function domain_access_node_grants($op, AccountInterface $account) {
+function domain_access_node_grants(AccountProxy $account, $op) {
   $grants = array();
   $active = domain_get_domain();
-  $id = $active->id();
+  $id = $active->getDomainId();
+
   // Grants for view are simple. Use the active domain.
   if ($op == 'view') {
     $grants['domain_id'][] = $id;
     return $grants;
   }
+
   // Grants for edit/delete require permissions.
-  $user_domains = domain_access_get_entity_values($account, DOMAIN_ACCESS_USER_FIELD);
-  if ($op == 'update' && $account->hasPermission('edit domain content') && isset($account_domains[$id])) {
-    $grants['domain_id'][] = $id;
-  }
-  if ($op == 'delete' && $account->hasPermission('delete domain content') && isset($account_domains[$id])) {
-    $grants['domain_id'][] = $id;
+  if($account->isAuthenticated()) {
```

Comment:
It is theoretically possible that someone would give this permission to an anonymous user. Imagine a scenario where there are two sites: notes.examplex.com and published.example.com. Anon users might be able to post / edit / delete on notes.example.com, with best content moving to published.example.com.

It's an odd edge-case, but I don't think we should be restrictive here by forcing a check for authentication.